gfx/drivers_context/wayland_ctx.c

```diff
@@ -607,6 +606,24 @@ static const struct wl_surface_listener wl_surface_listener = {
     nop,
 };
 
+static const struct wl_callback_listener frame_listener;
+
+static void frame_callback(void *data, struct wl_callback *callback, uint32_t time)
+{
+	gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
+	
+	if (callback)
+	  wl_callback_destroy(callback);
+	  
+	wl->frame_callback = wl_surface_frame(wl->surface);
```

Comment:
Why would you do this here?

Why are you not setting some flag that tells the rest of the software that it's ok to draw now if there happens to be something to update on screen on *this specific window*?

`wl_surface_frame()` should be called as part of the code path that has drawn a new frame and is just about to `wl_surface_commit()` to the compositor anyway. It is quite useless to just keep frame callbacks continuously firing if you don't actually synchronize your application drawing cycles to them. It also needs to be per each mapped `wl_surface`.